packages/core/src/result.ts

```diff
@@ -56,16 +56,16 @@ const DEFAULT_ON_KEYS = (keys: string[]): void => {}
  * The query result is the combination of the {@link ResultSummary} and
  * the array {@link Record[]} produced by the query
  */
-interface QueryResult {
-  records: Record[]
+interface QueryResult<Entries extends Dict = Dict> {
```

Comment:
nit: shouldn't it be called `Entry` everywhere?

Reply:
I kept the same naming convention already used in the Record

Reply:
Is this name exposed to the user or is it just internal? I was thinking this is a type and a type is usually singular, like you write `Dict` and not `Dicts`. But I don't mind either way. Just a nit pick here.

Reply:
It exposed in the auto-complete. I might call it `RecordShape` so.